### DISCUSSION TITLE
How to create block plugin using WordPress core blocks

### DISCUSSION THREAD

**Discussion by jagirbahesh** [WordPress/developer-blog-content#105] [Topic Ideas]
It is a commonly known fact that creating Gutenberg blocks can be quite complex, despite the abundance of online documentation. However, there are no clear instructions available on utilizing core blocks in our own Gutenberg block plugins.

Fortunately, WordPress offers numerous default blocks that can aid in plugin development and save a significant amount of time. The purpose of this blog is to provide a detailed, step-by-step guide on setting up a basic Gutenberg block plugin development process using core blocks.

In relation to this matter, I have previously conducted a session during the Ahmedabad WordPress Meetup where many individuals expressed their interest in obtaining related documentation and accessing the blog.

**Comment by bph** [comment]
@jagirbahesh - Thank you so much for starting this discussion! When you write "utilizing core blocks in our own Gutenberg block plugins" Could you elaborate a bit on that? Do you have an example use case? Or which part of the core blocks. 

There are [Block variations](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-variations/), and [Block styles](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-styles/) as well as [Block Templates](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-templates/) that could be used in plugin. 

There is also the [concept of InnerBlocks](https://developer.wordpress.org/block-editor/how-to-guides/block-tutorial/nested-blocks-inner-blocks/).

**Comment by jagirbahesh** [comment]
Hello @bph,

Sorry for late reply I have not received any notification for your message.
Let me explain more in details.

As you know that WordPress provide the default blocks like Text, Heading, Image and ETC.
we will use those default blocks in our custom Gutenberg plugin development.

Here I have shared the docs which I have already take this session in my local community.
https://speakerdeck.com/ahmedabadwordpress/using-core-blocks-create-gutenberg-blocks-plugin-in-wordpress-july-2022-ahmedabad-wordpress-meetup

Please check and let me know if you have any questions.

**Comment by bph** [comment]
@jagirbahesh thank you for clarifying it somewhat. If you review most of the tutorials published on this developer blog, they often provide a problem statement, an examples use case and then walk the reader through with step-by-step instructions through a solution, adding theoretical excursions along the way. A slide deck with a short code examples and many links doesn't convey, what a blog post would be about. 

It would easier to consider such a topic if you could provide a "[fat outline](https://bernoff.com/blog/fat-outline)" on how you would approach your article, and what a reader would accomplish following your tutorial.

**Comment by bph** [comment]
I close this after the discussion didn't progress.
